File: Sorting/BubbleSort/BubbleSort.py

def bubbleSort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        #Every iteration, last i items sorted
        for j in range(n-i-1):
            if(arr[j] > arr[j+1]):
                swapped = True
                arr[j], arr[j+1] = arr[j+1],arr[j]
        # One loop without swapping means that array already sorted
        if(not swapped):
            break
        

File: Sorting/BubbleSort/BubbleSort.java
package BubbleSort;

class BubbleSort {
    void sort(int arr[]) 
    { 
        int i,j,temp;
        int n = arr.length;
        boolean swapped;

        for(i = 0; i < n; i++){
            swapped = false;
            for(j = 0; j < n - i - 1; j++){
                if(arr[j] > arr[j + 1]){
                    temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }
            if(!swapped)
                break;
        }
    } 
  
    public static void main(String args[]) 
    { 
        /* Example Usage
        int arr[] = { 1, 6, 1, 3, 6, 8, 2, 4, 2, 3, 6, 9, 2, 5 ,0 ,7 ,3 ,5 }; 
  
        BubbleSort bubbleSort = new BubbleSort(); 
        
        bubbleSort.sort(arr); 

        for (int i = 0; i < arr.length; i++) 
            System.out.print(arr[i] + " "); 
        */
        
    } 
} 
